controllers/dataprotection/backup_controller.go

```diff
@@ -510,6 +510,16 @@ func (r *BackupReconciler) handleRunningPhase(
 		duration := request.Status.CompletionTimestamp.Sub(request.Status.StartTimestamp.Time).Round(time.Second)
 		request.Status.Duration = &metav1.Duration{Duration: duration}
 	}
+	if request.Spec.RetentionPeriod != "" {
+		// set expiration time
+		duration, err := request.Spec.RetentionPeriod.ToDuration()
+		if err != nil {
+			return r.updateStatusIfFailed(reqCtx, backup, request.Backup, fmt.Errorf("failed to parse retention period %s, %v", request.Spec.RetentionPeriod, err))
+		}
+		request.Status.Expiration = &metav1.Time{
```

Comment:
check if the duration.Seconds() greater than 0, if yes, set the expiration, else the retention period is 0, the backup will not be deleted. Like:

```
	duration, err := original.Spec.RetentionPeriod.ToDuration()
	if err != nil {
		return fmt.Errorf("failed to parse retention period %s, %v", original.Spec.RetentionPeriod, err)
	}
	if duration.Seconds() > 0 {
		request.Status.Expiration = &metav1.Time{
			Time: request.Status.StartTimestamp.Add(duration),
		}
	}
```